### DISCUSSION TITLE
How do I run Pesterchum on Linux?

### DISCUSSION THREAD

**Discussion by Syddney** [Dpeta/pesterchum-alt-servers#53] [Q&A]
I downloaded it onto my chromebook and have it in the linux emulator but can't figure out how to run it, how does it work?

**Comment by Dpeta** [comment]
Well, if you're able to run linux programs you should just be able to extract [``PesterchumAlt.-2.1.2-linux64-glibc2.28.tar.gz``](https://github.com/Dpeta/pesterchum-alt-servers/releases/tag/v2.1.2) and run the file called ``Pesterchum`` :0c

**Reply by Syddney** [reply]
how do I run the file though?

**Reply by Dpeta** [reply]
Well, on Linux you can just click the file & execute
![Screenshot_20210805_225916](https://user-images.githubusercontent.com/69427753/128420419-c42a99bd-a519-4224-9094-ef89000de677.png)
Or otherwise run it from the command line like this:
![Screenshot_20210805_230028](https://user-images.githubusercontent.com/69427753/128420591-9b477453-5658-4af3-92ca-dd89c83b2607.png)
I'm not sure how chromebooks work though :0c

**Reply by Syddney** [reply]
I'm not that good with tech :( how do I do the blue text in the command line?

**Reply by Dpeta** [reply]
Ah, that's valid :(
The blue text is just the folder I'm in rn.
Do you have a console/terminal/command line? :?c
If so you should be able to navigate to the folder you extracted the files with the ``cd`` command, and just run ``./Pesterchum`` once you're in the folder :0c

**Reply by Syddney** [reply]
everythings working except for the file itself, it wont recognise it being there :(

**Reply by Dpeta** [reply]
Are you certain you've navigated to the correct folder? :?c
You can use the ``ls`` command to list files in your current folder to check too :0c

**Reply by Syddney** [reply]
I'm in the correct folder, I checked and I found the pesterchum file, it won't recognise that the file is there though

**Reply by Dpeta** [reply]
Hm, are you sure the capitalization is correct? :?c
I think for that build it should be ``./Pesterchum`` and not ``./pesterchum``.

**Reply by Syddney** [reply]
Yeah I ran the command with the P capitalized

**Reply by Dpeta** [reply]
Ah alright-
I'm, not sure rn then- I've never used a chromebook before, so I don't really have much of a reference for most of this óvò
I asked the only person I know with a chromebook but she's never set up Pesterchum on it either

**Reply by Syddney** [reply]
darn, that sucks, thank you for trying to help though!

**Reply by Dpeta** [reply]
Np :0c
It ought to be possible somehow though, if you can run linux applications you should be able to run pchum. (Or if you can run Windows applications via Wine.)

**Comment by K-a-y-l-e-b** [comment]
I have a similar issue, also on a Chromebook. I'm using Linux for Linux compatible stuff and ChromeOS for Chrome compatible stuff.

My version of ChromeOS is currently Version 132.0.6834.190 (Official Build) (64-bit).

Some details about the Linux I'm using:
Virtualization: lxc
Operating System: Debian GNU/Linux 12 (bookworm)
Architecture: x86-64
Chassis: container ☐

I downloaded this: https://packages.debian.org/bookworm/libqt6multimedia6
and this: https://pypi.org/project/PyQt6/  to try and fufill the dependencies, and also fix the multimedia backend issue you described that crops up with Linux. 

I've confirmed that I'm in the correct folder, because I get the message you put there for Linux users to help them troubleshoot. 

_This_ message: 

QtMultimedia audio is a bit silly/goofy on Linux and relies on an appropriate backend being availible.
If it doesn't work, try installing your distro's equivalent of the qt6-multimedia-backend/qt6-multimedia/gstreamer packages.

I got _this_ message when in Terminal, in the Pesterchum folder, trying to use ./Pesterchum to run the app : 

The Wayland connection broke. Did the Wayland compositor die? 

_I_ think that the wayland connection is being a bit silly/goofy, myself. AND I think that's because of the silly/goofy nature of the kind of Linux container that is on Chromebooks. Many people with much more technology experience than me have told me that Linux on Chromebooks is extremELY silly/goofy (now I'm saying silly/goofy just to avoid cussing more at my SILLY AND GOOFY machine gRRRRRRRRRRRR) and I have experience + can confirm. Linux on Chromebooks is a nightmare. Not your fault whatsoever. It iis what iit iis. 

So, another thing:

 I was thinking about running Pesterchum from the python script to see if I could get around the wayland issue. The Linux on Chromebooks comes with a very very skimpy teeny tiny version of the current Python (which I think is Python 3?)  with some stuff missing to make it more lightweight. I made the computer get the FULL version of Python 3. I couldn't make it get pip. 
It gives me a message telling me that pip won't work in this environment, and forcing it to get pip will likely either break pip or the operating system. I scrounged around online and was told that usually Pipx is a valid alternative (especially for Chromebooks using Debian bookworm, because it won't hurt them)

Would it work in this specific scenario, though? Is there anything about Pipx or Pesterchum that would make them not work with each other? I have confirmed that Pipx works with my _computer_ just fine. It doesn't break anything or itself when used. But I want to make sure that it would be okay with Pesterchum.

**Reply by Dpeta** [reply]
I'm guessing it'd be fine if it can install PyQt correctly
